database-commons/src/main/java/co/cask/db/batch/source/DBSource.java

```diff
@@ -59,58 +58,49 @@
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
+import java.util.Objects;
 import java.util.Properties;
 import javax.annotation.Nullable;
 import javax.ws.rs.Path;
 
 /**
  * Batch source to read from a DB table
  */
-@Plugin(type = "batchsource")
-@Name("Database")
-@Description("Reads from a database table(s) using a configurable SQL query." +
-  " Outputs one record for each row returned by the query.")
-public class DBSource extends ReferenceBatchSource<LongWritable, DBRecord, StructuredRecord> {
+public abstract class DBSource extends ReferenceBatchSource<LongWritable, DBRecord, StructuredRecord>
+  implements JDBCDriverNameProvider {
+
   private static final Logger LOG = LoggerFactory.getLogger(DBSource.class);
 
-  private final DBSourceConfig sourceConfig;
-  private final DBManager dbManager;
-  private Class<? extends Driver> driverClass;
+  protected final DBSourceConfig sourceConfig;
+  protected Class<? extends Driver> driverClass;
 
   public DBSource(DBSourceConfig sourceConfig) {
     super(new ReferencePluginConfig(sourceConfig.referenceName));
     this.sourceConfig = sourceConfig;
-    this.dbManager = new DBManager(sourceConfig);
+  }
+
+  private static String removeConditionsClause(String importQuerySring) {
+    importQuerySring = importQuerySring.replaceAll("\\s{2,}", " ").toUpperCase();
+    if (importQuerySring.contains("WHERE $CONDITIONS AND")) {
```

Comment:
this may catch many of the common possibilities, but it is still possible to specify a query that will cause this to generate an invalid query.

Let's have a final  `else (importQueryString.contains("$CONDITIONS"))` and throw an exception that asks the user to remove the CONDITIONS clause when fetching the input schema.